apollo-kotlin-sample/src/main/java/com/apollographql/apollo/kotlinsample/commits/CommitsActivity.kt

```diff
@@ -0,0 +1,61 @@
+package com.apollographql.apollo.kotlinsample.commits
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.support.v7.widget.LinearLayoutManager
+import android.view.View
+import android.view.View.GONE
+import android.view.View.VISIBLE
+import com.apollographql.apollo.coroutines.toDeferred
+import com.apollographql.apollo.kotlinsample.GithubRepositoryCommitsQuery
+import com.apollographql.apollo.kotlinsample.KotlinSampleApp
+import com.apollographql.apollo.kotlinsample.R
+import com.apollographql.apollo.kotlinsample.repositories.CommitsAdapter
+import kotlinx.android.synthetic.main.activity_commits.*
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+
+class CommitsActivity : AppCompatActivity() {
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_commits)
+
+    val apolloClient = (application as KotlinSampleApp).apolloClient
+
+    val repoName = intent.getStringExtra(CommitsActivity.REPO_NAME_KEY)
+    supportActionBar?.title = repoName
+
+    val adapter = CommitsAdapter()
+
+    recyclerView.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
+    recyclerView.adapter = adapter
+
+    tvError.visibility = GONE
+    progressBar.visibility = VISIBLE
+    GlobalScope.launch(Dispatchers.Main) {
```

Comment:
I guess the recommended way would be use a scope tied to the activity lifecycle. That pull a bit of complexity in the sample though so I'm not sure it's worth it. I could cancel the job manually from `onDestroy` though to keep things confined.

Reply:
I guess the recommended way would be to use some coroutine scope tied to the activity lifecycle. This adds some complexity to the sample so I'm not sure it's worth it. 

I modified this code a bit to cancel the job from `onDestroy` so that the request is cancelled if the activity is destroyed.